packages/generator-single-spa/src/react/generator-single-spa-react.js

```diff
@@ -57,6 +43,35 @@ module.exports = class SingleSpaReactGenerator extends Generator {
         ])
       ).typescript;
     }
+
+    const packageJsonTemplate = await fs.readFile(
+      this.templatePath("package.json"),
+      { encoding: "utf-8" }
+    );
+    const packageJsonStr = ejs.render(packageJsonTemplate, {
+      packageManager: this.packageManager,
+      typescript: this.typescript,
+    });
+
+    const packageJson = JSON.parse(packageJsonStr);
+
+    if (this.typescript) {
+      // Will be added as a dependency via ts-package.json
+      delete packageJson.devDependencies["@types/jest"];
+      // Will be replaced by eslint-config-ts-react-important-stuff
+      delete packageJson.devDependencies["eslint-config-react-important-stuff"];
+    }
+
+    this.fs.extendJSON(this.destinationPath("package.json"), packageJson);
+
+    if (this.typescript) {
+      this.fs.extendJSON(
```

Comment:
I switched to using extendJSON and an external ts-package.json file for this, instead of inlining the dependencies within this file. The reason is that I couldn't avoid the double yarn install any other way

Reply:
Is this better than just inlining the TS dependencies in `package.json.ejs`, now that the latter is templated over `typescript` anyway?

Reply:
You could also take care of `      delete packageJson.devDependencies["eslint-config-react-important-stuff"];` that way

Reply:
Yeah I thought of doing it that way, and it would work. But then we couldn't run `yarn upgrade-interactive-latest --interactive` on the template package json to upgrade its dependencies. Which is pretty convenient for keeping things up-to-date. I decided with this approach for that reason, but agree this implementation is maybe less straightforward.